web/api/v1/api.go

```diff
@@ -263,11 +286,18 @@ func (api *API) query(r *http.Request) (interface{}, *apiError, func()) {
 		qs = stats.NewQueryStats(qry.Stats())
 	}
 
+	var remoteError *apiError
+	if fanout, ok := newQueryable.(storage.FanoutStorage); ok {
+		if fanout.HadRemoteError() {
```

Comment:
It's not actually being returned as an error, it's simply being used so that the wrap function can identify that there was a remote read error. It seemed like the simplest way to propagate that warning to the outermost json body. It feels like overkill to modify the apiFunc type to return an extra message when only a few methods may ever need it.

Reply:
It'd still be best to return the actual error that occurred back to the user

Reply:
Ok, in the case of multiple remote reads failing, should it be an array of all the error messages sent back?

Reply:
That's one way of doing it.

Reply:
Do you have a different suggestion?